src/transxchange/read.rs

```diff
@@ -378,90 +378,241 @@ impl Iterator for NaiveDateRange {
+                        }
+                        "Wednesday" => {
+                            regular_days.insert(Wed);
+                        }
+                        "Thursday" => {
+                            regular_days.insert(Thu);
+                        }
+                        "Friday" => {
+                            regular_days.insert(Fri);
+                        }
+                        "Saturday" => {
+                            regular_days.insert(Sat);
+                        }
+                        "Sunday" => {
+                            regular_days.insert(Sun);
+                        }
+                        "MondayToFriday" => {
+                            regular_days.insert(Mon);
+                            regular_days.insert(Tue);
+                            regular_days.insert(Wed);
+                            regular_days.insert(Thu);
+                            regular_days.insert(Fri);
+                        }
+                        "MondayToSaturday" => {
+                            regular_days.insert(Mon);
+                            regular_days.insert(Tue);
+                            regular_days.insert(Wed);
+                            regular_days.insert(Thu);
+                            regular_days.insert(Fri);
+                            regular_days.insert(Sat);
+                        }
+                        "MondayToSunday" => {
+                            regular_days.insert(Mon);
+                            regular_days.insert(Tue);
+                            regular_days.insert(Wed);
+                            regular_days.insert(Thu);
+                            regular_days.insert(Fri);
+                            regular_days.insert(Sat);
+                            regular_days.insert(Sun);
+                        }
+                        "NotSaturday" => {
+                            regular_days.insert(Mon);
+                            regular_days.insert(Tue);
+                            regular_days.insert(Wed);
+                            regular_days.insert(Thu);
+                            regular_days.insert(Fri);
+                            regular_days.insert(Sun);
+                        }
+                        "Weekend" => {
+                            regular_days.insert(Sat);
+                            regular_days.insert(Sun);
+                        }
+                        unknown_tag => {
+                            warn!("Tag '{}' is not a valid tag for DaysOfWeek", unknown_tag)
+                        }
+                    };
+                }
+            }
+        }
+        Ok(regular_days)
+    }
+    fn get_bank_holidays(
+        bank_holiday_operation: &Element,
+        bank_holidays: &HashMap<BankHoliday, Vec<Date>>,
+        validity_period: &ValidityPeriod,
+    ) -> Result<(HashSet<Date>, HashSet<Date>)> {
+        fn get_bank_holidays_dates(
```

Comment:
fn inception :scream: 
can you move the nested functions out of `generate_calendar_dates` function? it's difficult to read.
At least `get_regular_days` and `get_bank_holidays`

Reply:
OK, so I'm working on a big refactor of this all PR actually because I was not satisfied about that too. The idea will be to have a struct to save the information from `OperatingProfile` into a usable structure and only after, propose an iterator inside the validity period (probably replacing the `NaiveDateRange` with something else).